### PR TITLE
Add some more bail out cases for unused parameters and value assignments analyzers

### LINKED ISSUE #31744: "Remove unused parameter" wants to remove parameters used in an expression statement
``` C#
        public static IVsSolution CreateWithAdviseUnadviseSolutionEvents(uint adviseCookie)
        {
            var mock = new Mock<IVsSolution>();
            mock.Setup(x => x.AdviseSolutionEvents(It.IsAny<IVsSolutionEvents>(), out adviseCookie)).Returns(VSConstants.S_OK);
            mock.Setup(x => x.UnadviseSolutionEvents(It.IsAny<uint>())).Returns(VSConstants.S_OK);
            return mock.Object;
        }
```

Expected:

No message

Actual:
```
Severity	Code	Description	Project	File	Line	Category	Suppression State
Message	IDE0060	Remove unused parameter 'adviseCookie'	Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests	E:\project-system2\src\Microsoft.VisualStudio.ProjectSystem.Managed.VS.UnitTests\Mocks\IVsSolutionFactory.cs	31	Style	Active
```

### PR DESCRIPTION
**Pull request by mavasani**
1. Presence of expression trees
2. Assignment to VB static locals
3. Parameters of VB handles method
4. Parameters of MEF importing constructors

Fixes #31744

### CODE REVIEW

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs [hunk 1]
```diff
@@ -366,8 +390,15 @@ private void AnalyzeOperationBlockEnd(OperationBlockAnalysisContext context)
...
+
+                        // Bail out in following cases:
+                        //   1. End user has configured the diagnostic to be suppressed.
+                        //   2. Symbol has error type, hence the diagnostic could be noised
+                        //   3. Symbol is static. For example, assignment to static locals in VB
+                        //      is not unnecessary as the assigned value can be used on the next invocation.
                         if (_options.UnusedValueAssignmentSeverity == ReportDiagnostic.Suppress ||
-                            symbol.GetSymbolType().IsErrorType())
+                            symbol.GetSymbolType().IsErrorType() ||
+                            symbol.IsStatic)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs]: i'd prefer to actually check for teh explicit case of a 'static local'.

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs [hunk 2]
```diff
@@ -159,6 +159,10 @@ private static IEnumerable<INamedTypeSymbol> GetAttributesForMethodsToIgnore(Com
 
                 // Don't flag obsolete methods.
                 yield return compilation.ObsoleteAttribute();
+
+                // Don't flag MEF import constructors with ImportingConstructor attribute.
+                yield return compilation.SystemCompositionImportingConstructorAttribute();
+                yield return compilation.SystemComponentModelCompositionImportingConstructorAttribute();
```
[the remark this answers is not in the document]
Comment (mavasani) [reply src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs]: I agree, and in fact this is not the only analyzer that has to tackle this concern. https://github.com/dotnet/roslyn/issues/30172 raised a similar issue about incorrect uninitialized and unused fields flagged by compiler, and https://github.com/dotnet/roslyn/issues/31581#issuecomment-445024843 about unused private members used via reflection being flagged by our unused members analyzer.

IMO, the best approach to tackle this problem space is https://github.com/dotnet/roslyn/pull/31603, which enables frameworks/libraries to ship with custom diagnostic suppressors/filter as developer dependencies, and take away all such false positive burden from both the analyzer authors and end users of such libraries.

Tagging @tmat @jinujoseph - another analyzer that would benefit from programmatic suppressions.

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.BlockAnalyzer.cs]: Sure, will do.

**Comment by mavasani** [comment]
@CyrusNajmabadi addressed feedback.

@heejaechang @jinujoseph can you review? This will improve the dogfooding experience for project-system team.

**Comment by mavasani** [comment]
@heejaechang @CyrusNajmabadi - any further feedback? Thanks

File: src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (jinujoseph) [reply src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs]: @jaredpar  too for info

**Pull request merged**